waardepapieren-service/src/waardepapieren-service.js

```diff
@@ -90,10 +96,20 @@ class WaardenpapierenService {
     }
 
     let productClaim = await core.claim(nlxIdentity, resultArray)
+    
+    // Claim is now only identified by the attribute name (the need), there should be a check if the data is attested for the correct attribute
```

Comment:
Does this break the concept of using ephemeral-links (aka hashes) to anchor the data? As in, what is the problem exactly here?

Reply:
To be clair, that comment is about the created `ipv8PermLink`. The problem is that the attribute name is not unique. So when a new paper is requested the IPv8 attestation will be overriden with the new info.

The primary reason this problem exists is because we only use one wallet (peer) to store the IPv8 claims. A better situation would be if each identity (bsn) is effectively a IPv8 peer. However, as far as I know create new peers for IPv8 with the REST API is not possible.

The solution might be to use the BSN in/as the attribute name since it is both known by the site and waardepapieren-serivce. I am not sure if this is the right solution since attribute names are plublic so this would "leak" the BSN to the public

Reply:
Yeah, leaking the BSN is a big no-go. Maybe this is the time to see if we can actually make the ipv8 on mobile phone thing happen, since that is indeed the long game.